solr/core/src/java/org/apache/solr/spelling/SpellCheckCollator.java

```diff
@@ -96,7 +96,7 @@ public List<SpellCheckCollation> collate(
     while (tryNo < maxTries && collNo < maxCollations && possibilityIter.hasNext()) {
 
       if (queryLimits.maybeExitWithPartialResults("SpellCheck collator")) {
-        return List.of();
+        return new ArrayList<>();
```

Comment:
Might returning the `collations` so far (if any) be an alternative?
```suggestion
        return collations;
```

Reply:
Yes seems good. I’ve added these changes. Returning collations will provide partial spell-check results when the query limit is exceeded.